### PR TITLE
[DuckDB] Clean up windows special-casing

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

File: D/DuckDB/build_tarballs.jl [hunk 1]
```diff
@@ -14,10 +14,6 @@ sources = [
 script = raw"""
 cd $WORKSPACE/srcdir/duckdb*/
 
-if [[ "${target}" == *-mingw32 ]]; then
-    sed -i -E "/add_executable\(duckdb_rest_server server.cpp\)$/aif\(\$\{WIN32\}\)\n  set\(LINK_EXTRA -lwsock32 -lws2_32\)\nendif\(\)\n" tools/rest/CMakeLists.txt
-fi
-
```
Comment (giordano) [inline D/DuckDB/build_tarballs.jl:20]: I presume this was made unnecessary by https://github.com/duckdb/duckdb/commit/5f92c916d326a16af70d161a536ed797526c74ae?

File: D/DuckDB/build_tarballs.jl [hunk 2]
```diff
@@ -28,7 +24,6 @@ make -j${nproc}
 make install
 
 if [[ "${target}" == *-mingw32 ]]; then
-    mkdir -p ${libdir}
```
Comment (giordano) [inline D/DuckDB/build_tarballs.jl:31]: This doesn't hurt, it's actually useful, as it makes sure `${libdir}` is a directory: https://github.com/JuliaPackaging/Yggdrasil/pull/4170#issuecomment-1007568619.  I would _**not**_ remove it.

Comment (jeremiahpslewis) [reply D/DuckDB/build_tarballs.jl:31]: Based on your comment, I think appending /. is also a fix, right?

File: D/DuckDB/build_tarballs.jl (hunk 1 above)
Comment (jeremiahpslewis) [reply D/DuckDB/build_tarballs.jl:20]: I think that commit is not part of the 0.3.1 release; this commit is the one I believe addresses the issue: https://github.com/duckdb/duckdb/commit/86307233c4989ff1b538260a0f08ee1bb3fef0ef

File: D/DuckDB/build_tarballs.jl (hunk 2 above)
Comment (giordano) [reply D/DuckDB/build_tarballs.jl:31]: Yes, _**if**_ `${libdir}` is an existing directory in the first place, which `mkdir` ensures.

**Pull request merged**